### PR TITLE
Merge release-agua back to master.

### PR DESCRIPTION
**Pull request by fabian-guerra**
Merges the latest changes in v3.7.4 back to master.

### CODE REVIEW

File: platform/ios/ios.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -1635,6 +1645,16 @@
 			path = ../vendor/SMCalloutView;
 			sourceTree = "<group>";
 		};
+		DA8848911CBB049300AB86E3 /* reachability */ = {
```
Comment (friedbunny) [inline platform/ios/ios.xcodeproj/project.pbxproj]: These reachability references were removed in #10940 in `master` (and don’t need to be added back).

File: platform/darwin/src/MGLMapSnapshotter.mm [hunk 2]
```diff
@@ -115,182 +116,192 @@ - (void)startWithQueue:(dispatch_queue_t)queue completionHandler:(MGLMapSnapshot
...
-                NSError *error = [NSError errorWithDomain:MGLErrorDomain code:MGLErrorCodeSnapshotFailed userInfo:userInfo];
-                
-                // Dispatch result to origin queue
-                dispatch_async(queue, ^{
-                    completion(nil, error);
-                });
-            } else {
+            [strongSelf drawAttributedSnapshot:attributions snapshotImage:mglImage pointForFn:pointForFn queue:queue completionHandler:completion];
+        }
+        _snapshotCallback = NULL;
```
Comment (julianrex) [inline platform/darwin/src/MGLMapSnapshotter.mm]: Referencing an ivar in a block will retain self implicitly. I think we should replace these with:
```
if (strongSelf)
    strongSelf->_snapshotCallback = NULL;
```

File: platform/darwin/src/MGLMapSnapshotter.mm [hunk 3]
```diff
@@ -115,182 +116,192 @@ - (void)startWithQueue:(dispatch_queue_t)queue completionHandler:(MGLMapSnapshot
...
+    });
+    dispatch_async(queue, ^{
+        _mbglMapSnapshotter->snapshot(_snapshotCallback->self());
```
Comment (julianrex) [inline platform/darwin/src/MGLMapSnapshotter.mm]: Ditto.

File: platform/darwin/src/MGLMapSnapshotter.mm (hunk 2 above)
Comment (julianrex) [reply platform/darwin/src/MGLMapSnapshotter.mm]: Not sure if this is a merge-blocker though.

File: platform/darwin/src/MGLMapSnapshotter.mm [hunk 4]
```diff
@@ -90,6 +90,7 @@ @implementation MGLMapSnapshotter {
     std::unique_ptr<mbgl::MapSnapshotter> _mbglMapSnapshotter;
     std::unique_ptr<mbgl::Actor<mbgl::MapSnapshotter::Callback>> _snapshotCallback;
     NS_ARRAY_OF(MGLAttributionInfo *) *_attributionInfo;
```
Comment (julianrex) [inline platform/darwin/src/MGLMapSnapshotter.mm]: I'd prefer to see `attributionInfo` as a property. Any reason why it isn't?

**Pull request merged**